### ISSUE TITLE
 Potential dependency conflicts between bakerydemo and urllib3

### ISSUE THREAD

**Issue by NeolithEra** [wagtail/bakerydemo#243]
Hi, as shown in the following full dependency graph of  **_bakerydemo_**, elasticsearch==2.4.1 requires _**urllib3>=1.8, <2.0**_， while the installed version of **_requests_**(2.22.0) requires _**urllib3>=1.21.1,<1.26**_.

According to pip's _“first found wins”_ installation strategy, _**urllib3 1.25.3**_ is the actually installed version.  

Although the first found package version _**urllib3 1.25.3**_ just satisfies the later dependency constraint （_**urllib3>=1.21.1,<1.26**_), it will lead to a build failure once developers release a newer version of **_urllib3_**.

### Dependency tree--------
```
bakerydemo-master
| +-aws-requests-auth(version range:==0.4.0)
| | +-requests(version range:>=0.14.0)
| | | +-certifi(version range:>=2017.4.17)
| | | +-chardet(version range:<3.1.0,>=3.0.2)
| | | +-idna(version range:>=2.5,<2.9)
| | | +-urllib3(version range:<1.26,>=1.21.1)
| +-awscli(version range:==1.16.43)
| +-boto(version range:==2.45.0)
| +-botocore(version range:==1.12.33)
| | +-docutils(version range:>=0.10)
| | +-jmespath(version range:<1.0.0,>=0.7.1)
| +-dj-database-url(version range:==0.4.1)
| +-django(version range:>=2.1,<2.2)
| +-django-cache-url(version range:==2.0.0)
| +-django-dotenv(version range:==1.4.1)
| +-django-redis(version range:==4.8.0)
| | +-redis(version range:>=2.10.0)
| +-django-storages(version range:==1.6.5)
| +-elasticsearch(version range:==2.4.1)
| | +-urllib3(version range:>=1.8,<2.0)
| +-pillow(version range:==4.0.0)
| +-psycopg2(version range:==2.7.3.1)
| +-uwsgi(version range:==2.0.17.1)
| +-wagtail(version range:>=2.5,<2.6)
| +-wagtailfontawesome(version range:>=1.1.3,<1.2)
| | +-django(version range:>=1.7.1)
| | +-wagtail(version range:>=1.4.0)
| +-whitenoise(version range:==3.2.2)
```

Thanks for your attention.
Best,
Neolith

**Comment by NeolithEra** [comment]
### Solution
1.	Add urllib3<1.26,>=1.21.1 as a direct dependency in your requirements.txt to remove the conflict.

Build a good dependency ecosystem for python projects is our common goal ^_^.

@heymonkeyriot May I pull a request to solve this issue?

**Comment by gasman** [comment]
Hi @NeolithEra - thanks for the report. I wasn't aware of [this pip limitation](https://github.com/pypa/pip/issues/988) until now, but what you propose sounds correct. A pull request would be very welcome!

**Comment by gasman** [comment]
Fixed in #244
